data/src/client.rs

```diff
@@ -364,10 +374,19 @@ impl Client {
             Command::Numeric(RPL_LOGGEDIN, _) => {
                 log::info!("[{}] logged in", self.server);
             }
-            Command::PRIVMSG(_, _) | Command::NOTICE(_, _) => {
+            Command::PRIVMSG(channel, text) | Command::NOTICE(channel, text) => {
                 if let Some(user) = message.user() {
-                    // If we sent (echo) & context exists (we sent from this client), ignore
-                    if user.nickname() == self.nickname() && context.is_some() {
+                    // Highlight notification
+                    if message::reference_user(user.nickname(), self.nickname(), text)
+                        && self.highlight_blackout.allow_highlights()
+                    {
+                        events.push(Event::Notification(
+                            message.clone(),
+                            self.nickname().to_owned(),
+                            Notification::Highlight(user, channel.clone()),
+                        ));
```

Comment:
Now that we include encoded message here, you can directly return here and remove the let events = vec![]; above